wordpress-mobile/WordPress-iOS
# [Hackday] Design System Update: Colors | Fonts | Length | `DSButton`

## What's it all about?

We have worked with @osullivanchris during the Hack Day and all of this week to align on what the Design System should look like. This PR includes all those updates along with their representation in the Gallery. The Design System is separated into two categories: Foundation & Components. The former will contain things like colors and fonts while the latter will be all the common components we wish to use in our app.

The goal is to form a common protocol, language, between Design, Android and iOS. Design will mention the name of a Color, Font, Icon etc. that will exactly match the one in code. So as developers will not need to care about their underlying values. So a color like `#D67709` is simply `warning` for us. As the defined colors are dynamic, we don't need to worry about the dark theme `#E68B28` as it comes out of the box.

In the Components section, we only have the button right now. I have named it `DSButton` (subject to change) however it will have 3 Emhpasis styles (Primary, Secondary & Tertiary) and 3 Sizes (Large, Medium, Small). So this component covers  a total of 9 variations. You will also notice that they are not the usual Green background button. This is due to the new style we're adopting in the app and the green button will be replaced by this style going forward.

Components should include the interaction states and things like Haptic Feedback when necessary. So everyone can play around with them in the Gallery.

## Snapshot Reference
| Light Theme  | Dark Theme |
| ------------- | ------------- |
| ![Simulator Screenshot - iPhone 15 Pro - 2023-11-24 at 18 35 40](https://github.com/wordpress-mobile/WordPress-iOS/assets/15968946/9f036858-75ae-4153-aa5b-a1c687711365) | ![Simulator Screenshot - iPhone 15 Pro - 2023-11-24 at 18 35 47](https://github.com/wordpress-mobile/WordPress-iOS/assets/15968946/1094fdc2-77d7-4f21-a985-28034003c248) |
| ![Simulator Screenshot - iPhone 15 Pro - 2023-11-24 at 18 35 57](https://github.com/wordpress-mobile/WordPress-iOS/assets/15968946/b02565dd-e94b-42f3-925e-0a2364c33f31) | ![Simulator Screenshot - iPhone 15 Pro - 2023-11-24 at 18 36 00](https://github.com/wordpress-mobile/WordPress-iOS/assets/15968946/4a94b916-7c8e-47d9-9935-83407e152165) |
| ![Simulator Screenshot - iPhone 15 Pro - 2023-11-24 at 18 36 09](https://github.com/wordpress-mobile/WordPress-iOS/assets/15968946/0c659166-e4d3-4eb8-a6a2-0a91b50098d6) | ![Simulator Screenshot - iPhone 15 Pro - 2023-11-24 at 18 36 14](https://github.com/wordpress-mobile/WordPress-iOS/assets/15968946/406f9033-040c-4e48-98c1-8f9fa039a663) |

## Testing Steps
Run the app on Debug Mode and go to "Me -> App Settings -> Design System"
You will find "Foundation" & "Components" there.

To test the button Go to `Components > DSButton` and interact with the non-disabled versions there.
Primary should shrink and highlight on tap.
Secondary only shrinks.
Tertiary only highlights.

To test the haptic, please use a real device.

## Regression Notes
1. Potential unintended areas of impact
Because `DomainPurchaseChoicesView` used the formerly defined button, I have updated it with the new one. I have manually tested it an all is fine. I would appreciate it if one of you can test it too @salimbraksa @hassaanelgarem.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual test

3. What automated tests I added (or what prevented me from doing so)
We can consider adding snapshot testing to at least Design System and move all that into a separate package.

PR submission checklist:


UI Changes testing checklist:
